avm/src/daemon/daemon.rs

```diff
@@ -180,7 +195,98 @@ async fn run_agz_b64(agz_b64: &str) -> DaemonResult<()> {
   Ok(())
 }
 
-async fn get_daemon_props() -> Option<&'static DaemonProperties> {
+fn write_b64_file(file_name: &str, content: &str) -> DaemonResult<()> {
+  let pwd = env::current_dir();
+  match pwd {
+    Ok(pwd) => write(
+      format!("{}/{}", pwd.display(), file_name),
+      base64::decode(content).unwrap(),
+    )?,
+    Err(err) => {
+      let err = format!("{:?}", err);
+      return Err(err.into());
+    }
+  }
+  Ok(())
+}
+
+async fn get_files_b64() -> HashMap<String, String> {
+  let mut files_b64 = HashMap::new();
+  files_b64.insert("Dockerfile".to_string(), get_dockerfile_b64().await);
+  files_b64.insert("app.tar.gz".to_string(), get_app_tar_gz_b64().await);
+  files_b64
+}
+
+fn create_app_tar(content: Option<&String>) -> () {
+  if let Some(content) = content {
+    if let Err(err) = write_b64_file("app.tar.gz", &content) {
+      eprintln!("Could not create app.tar.gz. {:?}", err);
+      std::process::exit(1);
+    }
+  }
+}
+
+async fn generate_token() -> String {
+  let body = json!({
+    "deployConfig": deploy::get_config().await,
+  });
+  post_v1("localDaemonToken", body).await
+}
+
+async fn set_local_daemon_props(local_agz_b64: Option<String>) -> () {
+  let files_b64 = get_files_b64().await;
+  create_app_tar(files_b64.get(&"app.tar.gz".to_string()));
+  let agz_b64 = if let Some(local_agz_b64) = local_agz_b64 {
+    local_agz_b64
+  } else {
+    eprintln!(
+      "running the Alan Daemon in a local environment requires the \
+               --agz-file argument specifying the .agz file to run"
+    );
+    std::process::exit(1);
+  };
+  DAEMON_PROPS
+    .set(DaemonProperties {
+      clusterId: "daemon-local-cluster".to_string(),
+      agzB64: agz_b64,
+      deployToken: generate_token().await,
+      domain: "alandeploy.com".to_string(),
+      filesB64: files_b64,
+    })
+    .unwrap();
+}
+
+fn create_certs_if_local() -> () {
+  if ALAN_TECH_ENV.as_str() == "local" {
+    // Self signed certs for local dev
+    // openssl req -newkey rsa:2048 -nodes -keyout key.pem -x509 -days 365 -out certificate.pem -subj "/C=US/ST=California/O=Alan Technologies, Inc/CN=*.anycloudapp.com"
+    let mut open_ssl = std::process::Command::new("openssl")
+      .stdout(std::process::Stdio::null())
+      .stderr(std::process::Stdio::null())
+      .arg("req")
+      .arg("-newkey")
+      .arg("rsa:2048")
+      .arg("-nodes")
+      .arg("-keyout")
+      .arg("key.pem")
+      .arg("-x509")
+      .arg("-days")
+      .arg("365")
+      .arg("-out")
+      .arg("certificate.pem")
+      .arg("-subj")
+      .arg("/C=US/ST=California/O=Alan Technologies, Inc/CN=*.anycloudapp.com")
+      .spawn()
+      .expect("Error generating self signed certificate");
+    open_ssl.wait().expect("Failed to wait on child");
+  }
+}
```

Comment:
Not a blocker, but I think we may want to just use this in general, not just when local? Create the cert and key files if they don't exist, period, and then we simplify the cluster startup script.

Reply:
Indeed this would simplify cluster startup script but not sure if we should do it. Having the command on the startup script give us flexibility to change it easier and faster. What I will do here is create them if they do not exists and not based on `ALAN_TECH_ENV`.